ivy/functional/backends/tensorflow/random.py

```diff
@@ -26,6 +26,9 @@
 # ------#
 
 
+@with_supported_dtypes(
+    {"2.13.0 and below": ("float", "int32", "int64")}, backend_version
+)
```

Comment:
Use the following script to get the dtypes supported by this function:
```python
import tensorflow as tf

def get_supported_and_unsupported_dtypes():
    supported_dtypes = set()
    unsupported_dtypes = set()

    for dtype in [tf.float16, tf.float32, tf.float64, tf.int32, tf.int64,
                  tf.uint8, tf.uint16, tf.int8, tf.int16, tf.bool]:
        try:
            tf.random.uniform(shape=(1,), minval=0, maxval=1, dtype=dtype)
            supported_dtypes.add(dtype)
        except Exception:
            unsupported_dtypes.add(dtype)

    return supported_dtypes, unsupported_dtypes

if __name__ == "__main__":
    supported_dtypes, unsupported_dtypes = get_supported_and_unsupported_dtypes()

    print("Supported Data Types:")
    for dtype in supported_dtypes:
        print(dtype)

    print("\nUnsupported Data Types:")
    for dtype in unsupported_dtypes:
        print(dtype)
```
It will print 
```python
Supported Data Types:
<dtype: 'float32'>
<dtype: 'float64'>
<dtype: 'int32'>
<dtype: 'int64'>
<dtype: 'float16'>

Unsupported Data Types:
<dtype: 'uint8'>
<dtype: 'int16'>
<dtype: 'int8'>
<dtype: 'bool'>
<dtype: 'uint16'>
```

Reply:
that's great, if we add` tf.bfloat16` it will print it in the supported dtypes

Reply:
`Supported Data Types:
<dtype: 'float32'>
<dtype: 'float64'>
<dtype: 'int32'>
<dtype: 'int64'>
<dtype: 'bfloat16'>
<dtype: 'float16'>
Unsupported Data Types:
<dtype: 'uint8'>
<dtype: 'int16'>
<dtype: 'int8'>
<dtype: 'bool'>`

Reply:
> Supported Data Types:
<dtype: 'float32'>
<dtype: 'float64'>
<dtype: 'int32'>
<dtype: 'int64'>
<dtype: 'bfloat16'>
<dtype: 'float16'>
Unsupported Data Types:
<dtype: 'uint8'>
<dtype: 'int16'>
<dtype: 'int8'>
<dtype: 'bool'>

Reply:
Supported Data Types:
<dtype: 'float32'>
<dtype: 'float64'>
<dtype: 'int32'>
<dtype: 'int64'>
<dtype: 'bfloat16'>
<dtype: 'float16'>
Unsupported Data Types:
<dtype: 'uint8'>
<dtype: 'int16'>
<dtype: 'int8'>
<dtype: 'bool'>

Reply:
Awesome, let's add these dtypes to the `supported_dtypes` decorator then?

Reply:
it's already there the `float` in the decorator mean (`float16` `float32` `float64` `bfloat16`)

Reply:
Got it, makes sense. Thanks for explaining :sweat_smile: